src/containers/costume-tab.jsx

```diff
@@ -46,6 +48,17 @@ class CostumeTab extends React.Component {
         this.props.vm.deleteCostume(costumeIndex);
     }
 
+    handleUpdateSvg (svg, rotationCenterX, rotationCenterY) {
+        const storage = this.props.vm.runtime.storage;
+        const newAssetId = storage.builtinHelper.cache(
+            storage.AssetType.ImageVector,
+            storage.DataFormat.SVG,
+            (new TextEncoder()).encode(svg),
+        );
+        this.props.vm.editingTarget.sprite.costumes[this.state.selectedCostumeIndex].assetId = newAssetId;
+        this.props.vm.updateSvg(this.state.selectedCostumeIndex, svg, rotationCenterX, rotationCenterY);
+    }
```

Comment:
This is ok for now, but we should create an official API on the VM as soon as we can to avoid reaching into the VM like this.

Reply:
I moved this code to updateSvg in vm to clean up the interface. Now this PR depends on https://github.com/LLK/scratch-vm/pull/718